artiq/frontend/artiq_run.py

```diff
@@ -199,6 +199,7 @@ def run(with_file=False):
 
     try:
         exp_inst = _build_experiment(device_mgr, dataset_mgr, args)
+        exp_inst._HasEnvironment__argument_mgr.check_unprocessed_arguments()
```

Comment:
This should probably go into `_build_experiment` (splitting up the `return` statement), where you have `argument_mgr` around still.